index.js

```diff
@@ -1,55 +1,63 @@
-const express = require("express");
+const express = require('express');
 
-const PORT = 8080;
+const PORT = 8000;
 const app = express();
 
-var product = require('./src/data/products.json');
-var category = require('./src/data/categories.json');
+// Import methods to get productMap, categoryMap from dataService.js file.
+const dataService = require('./dataService');
 
-app.get('/', function (req, res) {
-    res.send(" ");
+// Create a route for app.
+app.get('/', function(req, res) {
+  res.send('');
 });
 
-app.get("/info", (req, res) => {
-    var serverInfo = require('./package.json');
-    res.json({serverName : serverInfo.name, serverVersion : serverInfo.version});
+// Display server information.
+app.get('/info', (req, res) => {
+  const serverInfo = require('./package.json');
+  res.json({serverName: serverInfo.name, serverVersion: serverInfo.version});
 });
 
-app.get("/products/all", (req, res) => {
-
-    // STEP 4. Combine category and product details, then send as JSON.
-
-    var combined = Object.assign({}, category, product);
-    res.json({allProducts : combined});
+// Send products and categories combined.
+app.get('/products/all', (req, res) => {
+  const combinedDataMap = dataService.getCombinedProductMap();
+  // Convert map into JSON object for output.
+  const combinedData = Object.fromEntries(combinedDataMap);
+  res.json({combinedData});
 });
 
-app.get("/product/:id", (req,res) => {
+// Get values for which product ID is a key.
+app.get('/product/:id', (req, res) => {
+  const productData = dataService.getProducts();
```

Comment:
Can't we use the `getCombinedProductMap`?

Reply:
Ohhh, you're right! Thanks for the suggestion.